btc-registration/src/main/kotlin/registration/btc/BtcRegistrationStrategyImpl.kt

```diff
@@ -30,8 +32,19 @@ class BtcRegistrationStrategyImpl(
             .flatMap { (addresses, takenAddresses) ->
                 try {
                     //It fetches all BTC addresses and takes one that was not registered
-                    val freeAddress = addresses.keys.first { btcAddress -> !takenAddresses.containsKey(btcAddress) }
-                    irohaAccountCreator.create(freeAddress, whitelist.toString().trim('[').trim(']'), name, pubkey)
+                    val freeAddress =
+                        addresses.first { btcAddress -> !takenAddresses.any { takenAddress -> takenAddress.address == btcAddress.address } }
```

Comment:
Possibly not thread safe, leave todo

Reply:
This is definitely not thread safe and will cause a lot of troubles. Thank you.